pkg/types/types.go

```diff
@@ -67,9 +67,10 @@ type NetConf struct {
 
 // Result is what gets returned from the plugin (via stdout) to the caller
 type Result struct {
-	IP4 *IPConfig `json:"ip4,omitempty"`
-	IP6 *IPConfig `json:"ip6,omitempty"`
-	DNS DNS       `json:"dns,omitempty"`
+	IP4 		*IPConfig `json:"ip4,omitempty"`
+	IP6 		*IPConfig `json:"ip6,omitempty"`
+	DNS 		DNS       `json:"dns,omitempty"`
+	CUSTOMATTR  	*CUSTOM    `json:"custom,omitempty"`
 }
```

Comment:
Two things here:

1. I think CUSTOMATTR should just be a dictionary, eg 'map[string]interface{}' in code.  That should marshal correctly to JSON an allow plugins to marshal their own structs into the dict.
2. It should just be named something like 'Args' or 'Extra' (no full capitalization required)
3. The JSON name should be the same as the Go struct member name, just without the initial capital.  Eg if the member is "Args" the JSON would be json:"args,omitempty".